pkg/landscaper/crdmanager/crdmanager.go

```diff
@@ -0,0 +1,186 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package crdmanager
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"path"
+	"time"
+
+	"github.com/gardener/landscaper/apis/config"
+
+	"github.com/go-logr/logr"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	embedFSCrdRootDir = "crdresources"
+	readerBufferSize  = 32
+)
+
+//go:embed crdresources/landscaper.gardener.cloud*.yaml
+var importedCrdFS embed.FS
+
+// CRDManager contains everything required to initialize or update Landscaper CRDs
+type CRDManager struct {
+	cfg          config.CrdManagementConfiguration
+	crdClientset *apiextensions.Clientset
+	log          logr.Logger
+	crdRawDataFS *embed.FS
+}
+
+// NewCrdManager returns a new instance of a CRDManager
+func NewCrdManager(log logr.Logger, mgr manager.Manager, lsConfig *config.LandscaperConfiguration) (*CRDManager, error) {
+	crdClientset, err := apiextensions.NewForConfig(mgr.GetConfig())
```

Comment:
I would work here with the `client.Client` package from the controller runtime to be consistent with the rest of the code.

Reply:
unfortunately, `client.Client` does not include direct access to scheme `apiextensions`. The `crdClienset` uses the client-go package and wraps it conveniently. Hence, we'd prefer to stick to it :)

Reply:
> unfortunately, client.Client does not include direct access to scheme apiextensions
it also only includes the "default" k8s types because they are defaulted. You can set the used schems in the clients option's:
```
import (
  apiextinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
 )
apiextensionsScheme := runtime.NewScheme()
apiextinstall.Install(apiextensionsScheme)
kubeClient, err := client.New(mgr.GetConfig(), client.Options{
  Scheme: apiextensionsScheme,
})
```

Reply:
thanks for the hint! Switch to client.Client has been pushed.